### DISCUSSION TITLE
Nested SCSS is not working for scoped SCSS.

### DISCUSSION THREAD

**Discussion by ankit-gupta1307** [vercel/next.js#14333] [Help] [answered]
# Bug report

## Describe the bug
I am writing the SCOPED Nested SCSS like -:

//index.module.scss
```
.container {
  background: #000;
  width: 100px;
  height: 100px;
  & > div{ 
     background: #FFCC00;
     width: 50px;
     height: 50px;
  }
}
```

//index.js
import classes from "./index.module.scss";
```
<div className={classes.container }>
   <div></div>
</div>
```

So, expected behaviour should be my inside div should be of different color and size but it's not taking any css. Please help.

This is a demo example.

**Comment by ankit-gupta1307** [comment]
@jamesmosier This is my use case. Please help how to use nested scss.

**Reply by ankit-gupta1307** [reply]
I will look into my configuration and will update here if I am able to fix the issue or not. Thanks much for replying.

**Comment by jamesmosier** [comment, marked as answer]
I'm not seeing this issue 🤔 I'm using Next `9.4.4`.

I used your exact scenario and was able to produce this:

<img width="1164" alt="Screen Shot 2020-06-18 at 8 29 25 PM" src="https://user-images.githubusercontent.com/2854919/85084321-72610280-b1a2-11ea-9341-d208c5641761.png">

**Comment by gabooh** [comment]
Have you found a solution ? I have the same problem, nested styles doesn't work here either.
With the following code, the .description blue background is not applied.

index.module.scss
```
.title {
  background-color: brown;

  .description {
    background-color: blue;
  }
}

```
index.js 
```
import styles from './index.module.scss'
[...]
<div className={styles.title}>
    Welcome to <a href="https://nextjs.org">Next.js!</a>

    <p className="description">
        Get started by editing <code>pages/index.js</code>
    </p>
</div>
```

And this is with a clean newly created project with create-next-app + npm i sass

**Reply by jamesmosier** [reply]
Your use case is different than the above one. The original one above should work.

You would have to write it as `<p className={styles.description}>` otherwise the SCSS module would have no idea to transform the child elements class.

Reference: https://www.javascriptstuff.com/css-modules-by-example/#example-4-nested-class-names

**Reply by ankitLV** [reply]
@gabooh if you have the nested classes then you will have to pick up from the styles object like @jamesmosier described above.

**Reply by ashfaqnisar** [reply]
Thanks for the reference @jamesmosier

**Comment by ankitLV** [comment]
@jamesmosier you can close this story, I was also foing the same thing like @gabooh.

Thanks for replies.

**Comment by gabooh** [comment]
Ok thank you, I understand it's not possible. That's a shame thought, would
be very useful and less verbose to be able to style child elements. It's
one of the best features of sass.

**Comment by mowais-ac** [comment]
I am also having the same problem, the styles are not applying to the nested classNames, anyone found a solution?

```
.test{border: solid 2px red;
    .test-nested{border: solid 4px yellow}
}
<div className="test">
    <div className="test-nested"></div>
</div>

**Comment by aestheticsdata** [comment]
same issue here. So what's the point to use scss module if we are forced to use it like a regular css module ?

**Reply by AlexisCatrix** [reply]
Personally, I use the way described in the sass basics section ( https://sass-lang.com/guide ) to compile my scss code to a css file, which allows me to use nested classes while rendering the result to the css file.

**Comment by tokyoshare** [comment]
You can use **:global** to make nested class work:

my-component.module.scss
 ```
.div {
      :global .x {
        color: red;
      }
    }
```

then you can use like this:

 ```
 import React from 'react'
    import styles from './my-component.module.scss'
    
    export default class MyComponent extends React.Component {
      public render() {
        return (
          <div className={styles.div}>
            <span className='x'>ololo</span>
          </div>
        )
      }
    }
```

**Reply by khuezy** [reply]
Thanks! The `:global` works great and is properly scoped within the parent container.
Eg:
```
//index.module.scss
.container :global {

  .blue { color: blue; }
  .green { color: green; }
   
}
```
```
import classes from "./index.module.scss";

<div className="blue">I'm not affected.</div>
<div className={classes.container }>
   <div className="blue">Hi I'm Blue</div>
   <div className="green">Hi I'm Green</div>
</div>
```

**Comment by ethyaan** [comment]
hey Folks,
I don't understand it.
why we should write scss in flat sequence??
then why i should `sass` or `scss` at all??
this nesting is the main reason to use scss.
and also using `:global` is not the right option, every time in dev mode we should restart the server to see the newly added classes

**Comment by Snouzy** [comment]
hey, you just have to tell nextJS the name of the class without carrying about the nest. 

For example : 
```scss
// Testimonial.module.scss

.testimonial { 
    border: 1px solid red;
    
    .nested-button {
         background: blue;
    }
}
```

```tsx
// Testimonial.jsx|tsx

import styles from './Testimonial.module.scss'

const Testimonial = () => (
    <div className={styles.testimonial}> {/* border will be 1px solid red */}
         <h1>Testimonial</h1>
         <button className={styles["nested-button"]}>View</button> {/* bg will be blue */}
    </div>
);

```
### Note : 
- take care that if you add `:global` in front of the nested classes, in dev mode, you'll have to hard reload every time.
- Of course, you can use the notation that you want (`styles.testimonial` or `styles["testimonial"]`) but be aware that in javascript, you'll have to access every dashed variable with quotes. 
So for the `nested-button` class, you have no choice : it will be `styles["nested-button"]`

`nextJS version : 12.2`
